packages/preferences/src/browser/views/preference-widget-bindings.ts

```diff
@@ -44,15 +51,37 @@ function createPreferencesWidgetContainer(parent: interfaces.Container): Contain
     child.bind(PreferencesTreeWidget).toSelf();
     child.rebind(TreeProps).toConstantValue({ ...defaultTreeProps, search: false });
     child.bind(PreferencesEditorWidget).toSelf();
-    child.bind(PreferencesDecorator).toSelf();
-    child.bind(PreferencesDecoratorService).toSelf();
-    child.rebind(TreeDecoratorService).toService(PreferencesDecoratorService);
 
-    child.bind(SinglePreferenceWrapper).toSelf();
     child.bind(PreferencesSearchbarWidget).toSelf();
     child.bind(PreferencesScopeTabBar).toSelf();
-    child.bind(SinglePreferenceDisplayFactory).toSelf();
     child.bind(PreferencesWidget).toSelf();
 
+    child.bind(PreferenceNodeRendererFactory).toFactory(({ container }) => (node: Preference.TreeNode) => {
+        const grandChild = container.createChild();
+        grandChild.bind(Preference.Node).toConstantValue(node);
+        if (Preference.LeafNode.is(node)) {
+            if (node.preference.data.enum) {
+                return grandChild.resolve(PreferenceSelectInputRenderer);
```

Comment:
Is it better to bind those classes being resolved in the parent container in transitive scope, and do a get here?

Reply:
Yes, that's a more flexible approach. I'll make the change. 👍